katago_server/distributed_efforts/tests/test_api.py

```diff
@@ -26,28 +42,68 @@ def test_get_job_anonymous(self):
+        response = client.post("/api/tasks/", {"git_revision":"abcdef123456abcdef123456abcdef1234567890"})
+        data = copy.deepcopy(response.data)
+        data["network"]["created_at"] = None # Suppress timestamp for test
+        data["run"]["id"] = None # Suppress id for test
+        assert str(data) == """{'kind': 'selfplay', 'run': {'id': None, 'url': 'http://testserver/api/runs/testrun/', 'name': 'testrun', 'data_board_len': 19, 'inputs_version': 7, 'max_search_threads_allowed': 8}, 'config': 'FILL ME', 'network': {'url': 'http://testserver/api/networks/testrun-randomnetwork/', 'run': 'http://testserver/api/runs/testrun/', 'name': 'testrun-randomnetwork', 'created_at': None, 'is_random': True, 'model_file': None, 'model_file_bytes': 0, 'model_file_sha256': '12341234abcdabcd56785678abcdabcd12341234abcdabcd56785678abcdabcd'}}"""
+        assert response.status_code == 200
+
+    def test_get_job_authenticated_valid_git_revision1_multipart(self):
+        client = APIClient()
+        client.login(username="test", password="test")
+        response = client.post("/api/tasks/", {"git_revision":"abcdef123456abcdef123456abcdef1234567890"},format="multipart")
+        data = copy.deepcopy(response.data)
+        data["network"]["created_at"] = None # Suppress timestamp for test
+        data["run"]["id"] = None # Suppress id for test
+        assert str(data) == """{'kind': 'selfplay', 'run': {'id': None, 'url': 'http://testserver/api/runs/testrun/', 'name': 'testrun', 'data_board_len': 19, 'inputs_version': 7, 'max_search_threads_allowed': 8}, 'config': 'FILL ME', 'network': {'url': 'http://testserver/api/networks/testrun-randomnetwork/', 'run': 'http://testserver/api/runs/testrun/', 'name': 'testrun-randomnetwork', 'created_at': None, 'is_random': True, 'model_file': None, 'model_file_bytes': 0, 'model_file_sha256': '12341234abcdabcd56785678abcdabcd12341234abcdabcd56785678abcdabcd'}}"""
+        assert response.status_code == 200
+
+    def test_get_job_authenticated_valid_git_revision2(self):
+        client = APIClient()
+        client.login(username="test", password="test")
+        response = client.post("/api/tasks/", {"git_revision":"1111222233334444555566667777888899990000"})
+        data = copy.deepcopy(response.data)
+        data["network"]["created_at"] = None # Suppress timestamp for test
+        data["run"]["id"] = None # Suppress id for test
+        assert str(data) == """{'kind': 'selfplay', 'run': {'id': None, 'url': 'http://testserver/api/runs/testrun/', 'name': 'testrun', 'data_board_len': 19, 'inputs_version': 7, 'max_search_threads_allowed': 8}, 'config': 'FILL ME', 'network': {'url': 'http://testserver/api/networks/testrun-randomnetwork/', 'run': 'http://testserver/api/runs/testrun/', 'name': 'testrun-randomnetwork', 'created_at': None, 'is_random': True, 'model_file': None, 'model_file_bytes': 0, 'model_file_sha256': '12341234abcdabcd56785678abcdabcd12341234abcdabcd56785678abcdabcd'}}"""
```

Comment:
Really nice test overhaul.
:)

Reply:
Still need to experiment with .save() and deal with tests with uploading networks and games though! :)